src/lib/utility.ts

```diff
@@ -103,3 +103,26 @@ export const generateCategoryItems = (categories: any[]) => {
 
   return items
 }
+
+export const combineAndSortPlaylistItems = (episodes: [any], mediaRefs: [any], itemsOrder: [string]) => {
+  const allPlaylistItems = [...episodes, ...mediaRefs]
+  const remainingPlaylistItems = [] as any[]
+  
+  const unsortedItems = allPlaylistItems.filter((x: any) => {
+    const isSortedItem = itemsOrder.some(id => x.id === id)
+    if (!isSortedItem) {
+      return x
+    } else {
+      remainingPlaylistItems.push(x)
+    }
+  })
```

Comment:
Kinda weird that his is a filter that might not return

Reply:
Agreed it's weird...not sure how to improve it though...

Reply:
Well since this is a filter and filter callback functions should return bools, I would suggest something like
`
const unsortedItems = allPlaylistItems.filter((x: any) => {
    const isSortedItem = itemsOrder.some(id => x.id === id)
    if (!isSortedItem) {
      return true
    } else {
      remainingPlaylistItems.push(x)
      return false
    }
  })
`